### PR TITLE
BCG stellar vs halo mass, BH mass vs. stellar mass, BH mass vel disp.

### PR DESCRIPTION
**Pull request by FHusko**
This PR makes the following three changes:

1. The stellar to halo mass relation plots for BCGs are created using recently added data.
2. The black hole - stellar mass relation for active and passive galaxies (from the simulations) are now plotted for centrals only, because the observational data by Terrazas et al. (2017) only includes what they consider centrals. This was agreed by Joop.
3. Added data to the BH mass vs. stellar LOS velocity dispersion plot (Saglia et al. 2016), also using recently added observational data.

### CODE REVIEW

**Review by MatthieuSchaller — APPROVED** [review]
That is fine by me but it is changing existing plots so I'll wait for Evgenii's approval too.

File: colibre/auto_plotter/black_holes.yml
Comment (EvgeniiChaikin) [inline colibre/auto_plotter/black_holes.yml:305]: I don't think the version of the submodule `velociraptor-comparison-data` in this repository has been updated recently. This means that the current version of `velociraptor-comparison-data` used here does not include the `Saglia+2016` data.

File: colibre/auto_plotter/stellar_mass_halo_mass.yml
Comment (EvgeniiChaikin) [inline colibre/auto_plotter/stellar_mass_halo_mass.yml:603]: Does the field `derived_quantities.projected_stellar_mass_10_kpc` actually exist? I don't see it defined in the registration file.

The same comment applies to `derived_quantities.projected_stellar_mass` measured in apertures of different size.

File: colibre/auto_plotter/stellar_mass_halo_mass.yml
Comment (EvgeniiChaikin) [inline colibre/auto_plotter/stellar_mass_halo_mass.yml:620]: In the caption, shouldn't it be Huang et al. (2022) instead of Huang et al. (2021)?

The same comment applies to captions of the other figures too.

File: colibre/auto_plotter/stellar_mass_halo_mass.yml
Comment (EvgeniiChaikin) [inline colibre/auto_plotter/stellar_mass_halo_mass.yml:664]: Has the submodule `velociraptor-comparison-data` been updated to include the BCG observational data, which was added recently?

**Review by EvgeniiChaikin — CHANGES_REQUESTED** [review]
The PR looks good to me but there are a couple of changes that need to be made before the PR is fully ready. I pointed them out below.

File: colibre/auto_plotter/black_holes.yml
Comment (FHusko) [reply colibre/auto_plotter/black_holes.yml:305]: Indeed. I wasn't entirely sure how this works, I thought that the submodule always points to the current master, but that clearly wasn't the case. It should be updated now.

File: colibre/auto_plotter/stellar_mass_halo_mass.yml
Comment (FHusko) [reply colibre/auto_plotter/stellar_mass_halo_mass.yml:603]: I had forgotten to actually add this (critical part), thanks for noticing! These derived projected stellar masses, which I have now added, are just averages over the three axes.

File: colibre/auto_plotter/stellar_mass_halo_mass.yml
Comment (FHusko) [reply colibre/auto_plotter/stellar_mass_halo_mass.yml]: Yes, I had been referring to this paper as being published in 2021 when I was starting to add these changes, but then when I found the full reference, the publication date was actually 2022. Clearly I didn't correctly update it everywhere.

File: colibre/auto_plotter/stellar_mass_halo_mass.yml
Comment (FHusko) [reply colibre/auto_plotter/stellar_mass_halo_mass.yml:664]: Should be done now.

**Comment by EvgeniiChaikin** [comment]
Thanks @FHusko! Now everything works as expected!

**Review by EvgeniiChaikin — APPROVED** [review]

**Pull request merged**
